Blurhash.ImageSharp/Encoder.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Linq;
+using System.Net.Mime;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Blurhash.Core;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing;
+
+namespace Blurhash.ImageSharp
+{
+    public class Encoder : CoreEncoder 
+    {
+        /// <summary>
+        /// Encodes a picture into a Blurhash string
+        /// </summary>
+        /// <param name="image">The picture to encode</param>
+        /// <param name="componentsX">The number of components used on the X-Axis for the DCT</param>
+        /// <param name="componentsY">The number of components used on the Y-Axis for the DCT</param>
+        /// <returns>The resulting Blurhash string</returns>
+        public string Encode(Image<Rgb24> image, int componentsX, int componentsY)
+        {
+            return CoreEncode(ConvertBitmap(image), componentsX, componentsY);
+        }
+        
+        /// <summary>
+        /// Encodes a picture into a Blurhash string
+        /// </summary>
+        /// <param name="image">The picture to encode</param>
+        /// <param name="componentsX">The number of components used on the X-Axis for the DCT</param>
+        /// <param name="componentsY">The number of components used on the Y-Axis for the DCT</param>
+        /// <returns>The resulting Blurhash string</returns>
+        public string Encode(Image<Rgba32> image, int componentsX, int componentsY)
+        {
+            return CoreEncode(ConvertBitmap(image), componentsX, componentsY);
+        }
+
+        /// <summary>
+        /// Converts the given bitmap to the library-independent representation used within the Blurhash-core
+        /// </summary>
+        /// <param name="sourceBitmap">The bitmap to encode</param>
+        internal static Pixel[,] ConvertBitmap<T>(Image<T> sourceBitmap) where T : struct, IPixel<T>
```

Comment:
Considering the implementation,  you should consume `Image<Rgb24>` or use proper conversion with `PixelOperations<T>.ToRgb24(...)`.